test/gtest/common/test_helpers.cc

```diff
@@ -276,6 +278,43 @@ scoped_setenv::~scoped_setenv() {
     }
 }
 
+ucx_env_cleanup::ucx_env_cleanup() {
+    const size_t prefix_len = strlen(UCS_CONFIG_PREFIX);
+    const char *var_name, *var_value;
+    char **envp, *saveptr, *envstr;
+
+    for (envp = environ; *envp != NULL; ++envp) {
+        envstr = ucs_strdup(*envp, "env_str");;
+
+        var_name = strtok_r(envstr, "=", &saveptr);
+        if (!var_name || strncmp(var_name, UCS_CONFIG_PREFIX, prefix_len)) {
+            ucs_free(envstr);
+            continue; /* Not UCX */
+        }
+
+        var_value = strtok_r(NULL, "=", &saveptr);
```

Comment:
why do we search for another '=' char? we should take the remainder of the string as-is
consider env var value with '=' inside

Reply:
@yosefe do we need this to call `putenv` instead? if yes, I've done it